### ISSUE TITLE
Add a toggle for parsing `noqa` comments

### ISSUE THREAD

**Issue by ragne** [ansible-community/ansible-lint#886]
##### Summary

The feature request arose when I tried to use ansible-lint as pre-commit hook and it took more than 10 seconds to complete.
Profiling shown that most of the time was spend in `ruamel.yaml.load` from `append_skipped_rules`. Which is perfectly logical, because `CLoader` AFAIK doesn't support comments. 
When I added a toggle flag, the total time went down to ~1.5 seconds.

Thus it would be nice to have a speed boost when parsing `#noqa` isn't needed for some reason (e.g you have skip tag instead)

##### Issue Type

- Feature Idea

##### Additional Information

Example flamegraph: 

<img width="994" alt="flamegraph" src="https://user-images.githubusercontent.com/21154977/87056229-02c4bd00-c20e-11ea-96ad-0bf895edeb45.png">

**Comment by ssbarnea** [comment]
I am very well aware of speed issues, and your case is not even major. I know repositories where it takes more than a minute to run. Still, disabling some essential features does not seem like the right fix.

Instead I would focus on making it run in parallel much better. With 8 cores would would likely make it ~6-7x faster, kinda similar with your idea, but more important: without disabling features.

I would recommend using https://opendev.org/zuul/zuul-jobs/src/branch/master/roles as reference model for performance measurement as it contains lots of roles.

**Comment by ragne** [comment]
Unfortunately during my attempts to test with zuul repo I was repeatedly beaten by:
 * too new version of ansible (of which you cannot attach __line__ anymore)
 * After merging your branch with fix I was facing a missing `zuul_return` module, which I don't know where to find

So, I used whatever handy repos I had. I'm sorry, I was unable to grasp where it would be really beneficial to introduce the parallelization (e.g on enumerating playbook files or go further and parallelize rule applying). I went with parallel run by file, which you can see in https://github.com/ansible/ansible-lint/compare/master...ragne:parallel

As for why I'm not bringing a PR, there's a couple of gotchas with my approach (and with parallelization in general). Either I did not figure it out, but `joblib` which I used refused to pickle any of rules (probably because of imports, didn't yet look), their "native" "loky" backend however chewed it fine and I was able to actually run 16 processes in parallel.
Now to the gotchas: I've noticed that decorating `parse_yaml_linenumbers` and extracting yaml loading from `_append_skipped_rules` would already give huge performance improvements. And _synchronising_ LRUCache across different processes is a bit difficult and less effective.

I still have to measure impact of parallelizing I've done, but after applying lru (I'm sure you might find a couple other bottlenecks) it isn't _so convincing anymore_
